### PR TITLE
Fix rectilinear X coordinates int edge case

### PR DESCRIPTION
**Pull request by akaszynski**
Encountered an edge case where edges are shown as triangles when the x spacing is int for a `pyvista.RectilinearGrid`:

```py

import pyvista
import numpy as np

xrng = np.arange(-10, 10, 5)
yrng = np.arange(-10, 10, 5).astype(float)
zrng = [1]
rec_grid = pyvista.RectilinearGrid(xrng, yrng, zrng)
rec_grid.plot(show_edges=True, cpos='xy')
```

### `main`

tmp

### This branch

image

### CODE REVIEW

File: pyvista/core/grid.py [hunk 1]
```diff
@@ -190,6 +190,10 @@ def _from_arrays(
         # Must at least be an x array
         if check_duplicates:
             raise_has_duplicates(x)
+
+        # weird side effects if x is not floating point
```
Comment (akaszynski) [inline pyvista/core/grid.py:194]:
```suggestion
        # edges are shown as triangles if x is not floating point
```

**Comment by adeak** [comment]
This is curious. Do you understand what happens?

The cells are `VTK_PIXEL` as expected, and extracting cell `0` gives me points in a sane (and same across main and this branch) order. And on both branches plotting that single cell with `show_edges=True` shows an ever so faint diagonal, much finer than what would be expected from proper edges. Yet this faint seam seems to be missing with `show_edges=False`. And again, this is the same for both branches, after extracting a single cell with `extract_cells(0)`.

File: pyvista/core/grid.py [hunk 2]
```diff
@@ -190,6 +190,10 @@ def _from_arrays(
         # Must at least be an x array
         if check_duplicates:
             raise_has_duplicates(x)
+
+        # edges are shown as triangles if x is not floating point
+        if not isinstance(x.dtype, np.floating):
```
Comment (adeak) [inline pyvista/core/grid.py:195]:
```suggestion
        if not np.issubdtype(x.dtype, np.floating):
```

**Review by adeak — APPROVED** [review]
LGTM, thanks

File: pyvista/core/grid.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adeak) [reply pyvista/core/grid.py]: Come to think of it, even if the bug only affects `x` we should convert every array for consistency...

Comment (akaszynski) [reply pyvista/core/grid.py]: For whatever reason this has messed up the windows tests...

**Review by adeak — DISMISSED** [review]
LGTM (again).

File: tests/plotting/test_plotting.py [hunk 3]
```diff
@@ -93,19 +93,20 @@
...
+    'test_plot_cell_data',
     'test_plot_complex_value',
-    'test_plot_helper_volume',
     'test_plot_helper_two_volumes',
+    'test_plot_helper_volume',
+    'test_plot_string_array',
+    'test_rectlinear_edge_case',
+    'test_scalars_by_name',
+    'test_user_annotations_scalar_bar_volume',
+    'test_user_annotations_scalar_bar_volume',
```
Comment (tkoyama010) [inline tests/plotting/test_plotting.py:109]: Duplication.
```suggestion
    'test_user_annotations_scalar_bar_volume',
```

File: tests/plotting/test_plotting.py [hunk 4]
```diff
@@ -93,19 +93,20 @@
...
-    'test_enable_stereo_render',  # occurs even without Windows OSMesa
-    'test_plot_add_scalar_bar',
-    'test_plot_cell_data',
-    'test_scalars_by_name',
-    'test_user_annotations_scalar_bar_volume',
-    'test_plot_string_array',
     'test_cmap_list',
     'test_collision_plot',
     'test_enable_stereo_render',
+    'test_enable_stereo_render',
```
Comment (tkoyama010) [inline tests/plotting/test_plotting.py:99]: Duplication.
```suggestion
```

**Review by tkoyama010 — CHANGES_REQUESTED** [review]
Thanks for fixing this. I have small suggestions.

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**